kakarot_scripts/compile_kakarot.py

```diff
@@ -22,8 +31,27 @@ def main():
     # %% Compile
     logger.info(f"ℹ️  Compiling contracts for network {NETWORK['name']}")
     initial_time = datetime.now()
+
+    # Split contracts into Cairo 0 and Cairo 1 to avoid
+    # re-compiling the same package multiple times.
+    cairo0_contracts = []
+    cairo1_packages = set()
```

Comment:
Why a list for one and a set for the other ?

Reply:
you compile cairo0 contracts individually but you compile cairo1 contracts per package. Thus you only want packages to be represented once, even if a package contains multiple contracts.